core/domain/suggestion_services.py

```diff
@@ -2904,3 +2908,334 @@ def enqueue_contributor_ranking_notification_email_task(
+        <div style="display:block; position: absolute; top: 80%; 
+        left: 50%; transform: translate(-50%, -50%); width: 100%;">
+        <div class="signature" style="width: 50%; float:left; 
+        text-align: center;"><div style="
+        font-family: Brush Script MT, Brush Script Std, cursiv;
+        font-size: 36px; line-height: 10px; margin-top: 0px;
+        color: #000000;">"""+ signature +"""</div> <center> 
+        <p class="line" style="width: 186px; height: 0px; border: 1px 
+        solid #00645C;"></p> </center> <p style="font-family: 'Roboto'; 
+        font-style: normal; font-weight: 400; font-size: 24px; 
+        line-height: 28px; margin-top: 0px; color: #8F9899;">SIGNATURE</p> 
+        </div> <div class="date" style="width: 50%; float:left; 
+        text-align: center;"><div style="font-family: 'Roboto';
+        font-size: 24px; color: #000000; line-height: 10px;">
+        """ + date + """</div><center> <p class="line" style="
+        width: 186px; height: 0px; border: 1px solid #00645C;"></p> 
+        </center> <p style="font-family: 'Roboto'; font-style: normal; 
+        font-weight: 400; font-size: 24px; line-height: 28px; 
+        margin-top: 0px; color: #8F9899;">DATE</p> </div> </div> </div>
+        </div> </div> </div> </body> </html>
+    """
+
+    return translation_submitter_certificate_template
+
+
+def _generate_question_contributor_certificate(
+    from_date: datetime.datetime,
+    to_date: datetime.datetime,
+    username: str,
+    user_id: str,
+    date: str
+) -> str:
+    """Generates a certificate for contributors' question contributions.
+
+    Args:
+        suggestion_type: str. The type of the suggestions that the
+            certificate needs to be generated.
+        from_date: datetime.datetime. The first date that the contributions
+            should be considered for the certificate.
+        to_date: datetime.datetime. The last date that the contributions
+            should be considered for the certificate.
+        username: str. The username of the contributor.
+        user_id: str. The user ID of the contributor.
+        date: datetime.datetime. The current date that should be included in the
+            certificate.
+
+    Returns:
+        str. The HTML template to generate the certificate.
+
+    Raises:
+        Exception: The suggestion type given to generate the certificate is
+            invalid.
+    """
+    signature = feconf.QUESTION_TEAM_LEAD
+
+    all_sugestions = (
+        suggestion_models.GeneralSuggestionModel
+            .get_question_suggestions_submitted_before_given_dates(
+                to_date, from_date, user_id))
+
+    minutes_contributed = 0
+    for model in all_sugestions:
+        suggestion = get_suggestion_from_model(model)
+
+        content = suggestion.change.question_dict[
+            'question_state_data']['content']['html']
+
+        if 'oppia-noninteractive-image' in content:
```

Comment:
Can you link the doc in a comment where this formula was derived?

Reply:
https://docs.google.com/spreadsheets/d/1ykSNwPLZ5qTCkuO21VLdtm_2SjJ5QJ0z0PlVjjSB4ZQ/edit?usp=sharing

Reply:
I meant link the doc in a comment in the code, so that future readers understand how this formula was derived and what it means. Also, do you have any supporting documentation with the reasoning?